qpc/report/merge.py

```diff
@@ -41,25 +44,129 @@ def __init__(self, subparsers):
+        group.add_argument('--job-ids', dest='scan_job_ids', nargs='+',
+                           metavar='SCAN_JOB_IDS', default=[],
+                           help=_(messages.REPORT_SCAN_JOB_IDS_HELP))
+        group.add_argument('--report-ids', dest='report_ids', nargs='+',
+                           metavar='REPORT_IDS', default=[],
+                           help=_(messages.REPORT_REPORT_IDS_HELP))
+        group.add_argument('--json-files', dest='json_files', nargs='+',
+                           metavar='JSON_FILES', default=[],
+                           help=_(messages.REPORT_JSON_FILE_HELP))
+        self.json = None
+        self.report_ids = None
+
+    @staticmethod
+    def get_id_and_sources(filename):
+        """Return the sources and id from a valid json details report file.
+
+        :param filename: the filename to read
+        :returns: the id and list of sources found
+        :raises: ValueError if incoming value is not a file that could be found
+        """
+        input_path = os.path.expanduser(os.path.expandvars(filename))
+        if os.path.isfile(input_path):
+            try:
+                with open(input_path, 'r') as in_file:
+                    result = in_file.read()
+                    report_id = json.loads(result).get('id')
+                    sources = json.loads(result).get('sources')
+                    if not report_id or not sources:
+                        print(_(messages.REPORT_INVALID_JSON_FILE % filename))
+                        sys.exit(1)
+            except EnvironmentError as err:
+                err_msg = (messages.READ_FILE_ERROR % (input_path, err))
+                print(err_msg)
+                sys.exit(1)
+            return report_id, sources
+        else:
+            raise ValueError(_(messages.NOT_A_FILE % input_path))
+
+    def _get_report_ids(self):
+        """Grab the report ids from the scan job if it exists.
+
+        :returns Boolean regarding the existence of scan jobs &
+        the report ids
+        """
+        not_found = False
+        report_ids = []
+        for scan_job_id in set(self.args.scan_job_ids):
+            # check for existence of scan_job
+            path = SCAN_JOB_URI + str(scan_job_id) + '/'
+            response = request(parser=self.parser, method=GET,
+                               path=path,
+                               params=None,
+                               payload=None)
+            if response.status_code == codes.ok:  # pylint: disable=no-member
+                json_data = response.json()
+                report_id = json_data.get('report_id', None)
+                if report_id:
+                    report_ids.append(report_id)
+                else:
+                    # there is not a report id associated with this scan job
+                    print(_(messages.SCAN_JOB_REPORT_ID_DOES_NOT_EXIST %
+                            scan_job_id))
+                    not_found = True
```

Comment:
You could collect the scanjobs ids that were not found instead of a boolean and print a single message.  

Something like:
```
if bool(invalid_scanjob_ids):
   print('The following scanjobs ids have no report: 1,2,3')
```

Reply:
I am fine with this but there could possibly be two messages because some reports don't exist and some scan jobs don't exist. It will still be better than it currently is!